zerver/data_import/import_util.py

```diff
@@ -730,8 +728,18 @@ def process_emojis(
 
         emoji_records.append(emoji_record)
 
-    # Run downloads in parallel
-    run_parallel_wrapper(partial(get_emojis, emoji_dir), upload_emoji_list, threads=threads)
+        # Directly download the emoji and patch the file_name with the correct extension
+        # based on the content-type returned by the server. This is needed because Slack
+        # sometimes returns an emoji url with .png extension despite the file being a gif.
+        content_type = get_emojis(emoji_dir, emoji_url, emoji_path)
+        if content_type is not None:
```

Comment:
So if we don't get a content-type, then ...?  I'm not sure what happens here, but it feels like it'd be easiest to reason about if it were "log and skip the emoji"

Reply:
Wouldn't the most obvious approach be log and keep the emoji, allowing the type to get inferred by the file extension? If for whatever reason the API doesn't specify the content type header, but it told us the emoji url is "xxxx.png", just treat it as a `.png`?

Reply:
Suuuuure.  Mostly I'm hesitant because part of this work has been to remove as many content-type heuristics that get us into trouble as we can.  But it seems probably-worth-it in terms of "be liberal in what you accept."

I think I'd argue that that heuristic logic should go into `get_emojis`, then.

Also probable that we should do the same check here as we do in `check_add_realm_emoji`:
```python3
    # This mirrors the check in upload_emoji_image because we want to
    # have some reasonable guarantees on the content-type before
    # deriving the file extension from it.
    if content_type not in THUMBNAIL_ACCEPT_IMAGE_TYPES or content_type not in INLINE_MIME_TYPES:
        raise BadImageError(_("Invalid image format"))
```

Reply:
> I think I'd argue that that heuristic logic should go into get_emojis, then.

That'd make sense, but the logic that happens here in the `content_type is None` case is "do nothing" (because we just keep the original file extension). Then I thought that `get_emoji` could just do `guess_type` on the URL and return that implied `content_type` (which would then get used to set the `png` extension like it was originally, so a noop effectively). But this exact logic is already done when setting the file extensions at the very beggining, in `build_realmemoji`:
https://github.com/zulip/zulip/blob/114f13e0ee0ca0ed053837de9889a11dee636263/zerver/data_import/slack.py#L224-L232

So seems super weird to re-do the same thing here, just to end in a noop.

Added the `check content_type is an image format` logic suggested though.